src/resolve/resolved-bus.c

```diff
@@ -1208,7 +1226,7 @@ static int resolve_service_hostname(DnsQuery *q, DnsResourceRecord *rr, int ifin
         if (r < 0)
                 return r;
 
-        r = dns_query_new(q->manager, &aux, question, question, NULL, ifindex, q->flags|SD_RESOLVED_NO_SEARCH);
+        r = dns_query_new_for_bus(q->manager, &aux, question, question, NULL, ifindex, q->flags|SD_RESOLVED_NO_SEARCH, NULL);
         if (r < 0)
                 return r;
```

Comment:
Setting `NULL` for `error` in `dns_query_new_for_bus()` is meaningless. Please use `dns_query_new()` and `reply_method_errorf()` on failure.
```c
r = dns_query_new(...);
if (r == -ENOANO)
        return reply_method_errorf(q, BUS_ERROR_DNS_REFUSED, "DNS query type refused.");
if (r < 0)
        return r;
```